### PR TITLE
[react-jsonschema-form] fix missing types on `onFocus`

### PR DESCRIPTION
**Pull request by medfreeman**
and `onBlur` second argument in `WidgetProps` interface

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes:
-  https://github.com/mozilla-services/react-jsonschema-form/tree/10a6b64918310576b0d54cdc28848cad016a75ac/src/components/widgets
- 
 https://github.com/DefinitelyTyped/DefinitelyTyped/blob/813999bd2b6b2c75dec22b5bba0b52c853af787c/types/react-jsonschema-form/index.d.ts#L15
-  https://github.com/DefinitelyTyped/DefinitelyTyped/blob/813999bd2b6b2c75dec22b5bba0b52c853af787c/types/json-schema/index.d.ts#L223

Basically the addition corresponds to any widget value possible types, which equals to `JSONSchema6Type` minus `array` and `object`, since a widget only deals with a single value at its scope.

- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by sbusch — APPROVED** [review]
The change looks good to me. The tests aren't verifying that much but are sufficient until RJSF has more formally defined widgets/fields.
